sync3/sort.go

```diff
@@ -145,10 +147,10 @@ func (s *SortableRooms) comparatorSortByName(i, j int) int {
 
 func (s *SortableRooms) comparatorSortByRecency(i, j int) int {
 	ri, rj := s.resolveRooms(i, j)
-	if ri.LastInterestedEventTimestamp == rj.LastInterestedEventTimestamp {
+	if ri.GetLastInterestedEventTimestamp(s.listKey) == rj.GetLastInterestedEventTimestamp(s.listKey) {
 		return 0
 	}
-	if ri.LastInterestedEventTimestamp > rj.LastInterestedEventTimestamp {
+	if ri.GetLastInterestedEventTimestamp(s.listKey) > rj.GetLastInterestedEventTimestamp(s.listKey) {
```

Comment:
It would be nice to precache this to avoid 2x map hits, i.e `tsRi := ri.GetLastInterestedEventTimestamp(s.listKey)` and `tsRj := rj.GetLastInterestedEventTimestamp(s.listKey)` then if `tsRi == rsRj` etc

Reply:
I was hoping that the compiler would be smart enough to do this for me. But I guess it has absolutely no idea what side effects the called function has. And even if it was just a plain lookup, it has no idea about what data races are going on behind the scenes.